### ISSUE TITLE
Portlets aren't styled properly

### ISSUE THREAD

**Issue by mwoehlke-kitware** [jenkinsci/dashboard-view-plugin#326]
### Jenkins and plugins versions report

<details>
<summary>Environment</summary>

```text
Dashboard View 2.495.v07e81500c3f2
```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

I have no idea what this is asking.

### Reproduction steps

Add a "Jobs Grid" or (AFAICT) any other portlet to any view. 

### Expected Results

The portlet should look like native tables (e.g. the "All" view) and should respect styling (e.g. dark mode).

### Actual Results

"Tables" are a mixture of a `div` for the header (which should be `thead` / `th`) and use hard-coded inline CSS. In e.g. dark themes, the borders are unnecessarily high-contrast and headers are effectively illegible.

### Are you interested in contributing a fix?

I think I have a fix, but am not sure how to test it:
```diff
diff --git a/src/main/resources/hudson/plugins/view/dashboard/DashboardPortlet/index.jelly b/src/main/resources/hudson/plugins/view/dashboard/DashboardPortlet/index.jelly
index 98dee36..169911e 100644
--- a/src/main/resources/hudson/plugins/view/dashboard/DashboardPortlet/index.jelly
+++ b/src/main/resources/hudson/plugins/view/dashboard/DashboardPortlet/index.jelly
@@ -3,6 +3,8 @@
   <j:set var="items" value="${it.dashboard.items}"/>
   <j:set var="jobs" value="${it.dashboard.jobs}"/>
   <l:layout title="${it.dashboard.displayName} - ${it.displayName}">
+    <link href="${rootURL}/plugin/dashboard-view/css/portlet.css" type="text/css"
+      rel="stylesheet"/>
     <link href="${rootURL}/plugin/dashboard-view/css/full_screen.css" type="text/css"
       rel="stylesheet"/>
     <l:main-panel>
diff --git a/src/main/resources/hudson/plugins/view/dashboard/core/JobsPortlet/portlet.jelly b/src/main/resources/hudson/plugins/view/dashboard/core/JobsPortlet/portlet.jelly
index a5e87a1..c186f90 100644
--- a/src/main/resources/hudson/plugins/view/dashboard/core/JobsPortlet/portlet.jelly
+++ b/src/main/resources/hudson/plugins/view/dashboard/core/JobsPortlet/portlet.jelly
@@ -11,7 +11,7 @@ SPDX-License-Identifier: MIT
     <j:forEach var="row" items="${it.jobs}">
       <tr>
         <j:forEach var="job" items="${row}">
-          <td style="border: 1px solid #bbb;">
+          <td>
             <j:if test="${job.buildable and job.hasPermission(job.BUILD)}">
               <j:choose>
                 <j:when test="${job.parameterized}">
diff --git a/src/main/resources/hudson/plugins/view/dashboard/decorate.jelly b/src/main/resources/hudson/plugins/view/dashboard/decorate.jelly
index e8209c0..7f08271 100644
--- a/src/main/resources/hudson/plugins/view/dashboard/decorate.jelly
+++ b/src/main/resources/hudson/plugins/view/dashboard/decorate.jelly
@@ -37,7 +37,7 @@ THE SOFTWARE.
     </st:attribute>
   </st:documentation>
   <div align="center">
-    <div id="title-${attrs.portlet.id}" align="left" style="border: 1px solid #999999; background-color: #F0F0F0; padding: 4px; font-weight: bold;">
+    <div id="title-${attrs.portlet.id}" align="left" style="background-color: var(--table-background); border-top-left-radius: calc(var(--table-border-radius) + 2px); border-top-right-radius: calc(var(--table-border-radius) + 2px); padding: 6px 12px 6px 20px; font-weight: bold;">
       ${attrs.portlet.displayName}
       <a href="${attrs.portlet.url}">
         <img id="cmdMax-${attrs.portlet.id}"
@@ -54,7 +54,7 @@ THE SOFTWARE.
         alt="${%Click to expand portlet}" tooltip="${%Click to expand portlet}"
         style="visibility: hidden; display: none;" width="16" height="16" />
     </div>
-  <table class="pane" id="${attrs.portlet.id}" style="margin-top: 0px; border-top: none;">
+  <table class="jenkins-table portlet-table" id="${attrs.portlet.id}" style="margin-top: 0px; border-top: none; border-top: none; border-top-left-radius: 0; border-top-right-radius: 0;">
     <d:invokeBody />
   </table>
   </div>
diff --git a/src/main/webapp/css/portlet.css b/src/main/webapp/css/portlet.css
new file mode 100644
index 0000000..67b0430
--- /dev/null
+++ b/src/main/webapp/css/portlet.css
@@ -0,0 +1,9 @@
+.jenkins-table.portlet-table {
+    border-spacing: 2px 2px;
+    border-width: 3px;
+}
+
+.jenkins-table.portlet-table > tbody > tr > td {
+    height: auto;
+    border-radius: var(--table-row-border-radius);
+}
```

I'm also not confident this is the right way to add the new `.css`.

**Comment by svenevs** [comment]
To clarify, we enabled [the dark theme](https://plugins.jenkins.io/dark-theme/), and it appears that this plugin emits `<td>` headings with hard-coded light backgrounds.

It could be that the appropriate fix is to drop coloring altogether (relying on jenkins theming more directly).  Or alternatively, give some user-specific styling options for the portlets as is done for left / right portlet size?

![dashboard_color_01](https://github.com/jenkinsci/dashboard-view-plugin/assets/5871461/4f871fdd-0469-49bb-a97e-a14a6ba2d82d)

FWIW our Jenkins controller is running version 2.426.2, and a sample table (tell your browser to flip between auto / light / dark modes to see): https://drake-jenkins.csail.mit.edu/view/Nightly%20Production/

Dark mode (hard to read):

![dashboard_color_dark](https://github.com/jenkinsci/dashboard-view-plugin/assets/5871461/d58c68c4-d7b7-47f1-ab08-d4011d282483)

Light mode (legible):

![dashboard_color_light](https://github.com/jenkinsci/dashboard-view-plugin/assets/5871461/2f9f9e26-1c38-4012-a9c9-a467d6e321f3)

It affects pretty much every portlet you can add via the plugin, and removing the hard-coded styling seems like the right fix (?)  We can also implement a server-side hack to modify it on our end as well.

**Comment by mwoehlke-kitware** [comment]
Pedantic: the headings are actually `<div>`s; see proposed patch for where they come from. I don't think user (i.e. site) styling using hard-coded colors is the right answer; IIUC, dark mode may or may not be enabled depending on user preferences, which means any value you could configure is going to be wrong for either light mode or dark mode. "Use the theme styling" seems like TRTTD.

BTW, here is what it (theoretically) looks like with the patch:
![preview](https://github.com/jenkinsci/dashboard-view-plugin/assets/6474409/08b5ce70-b857-451e-83a8-1031e55a2ec6)

**Comment by TobiX** [comment]
It would probably be best to follow the [Jenkins Design Library](https://weekly.ci.jenkins.io/design-library/) with this plugin... (And remove all hard coded styles from this plugin)

**Comment by mwoehlke-kitware** [comment]
> It would probably be best to follow the [Jenkins Design Library](https://weekly.ci.jenkins.io/design-library/) with this plugin

I think that's roughly what I'm trying to do with the proposed patch. The reason there is still some deviation / customization is a) in order to retain being able to collapse and/or full-screen portlets, and b) because the the DL thinks "tables" are one-item-per-row with columns representing various fields, whereas the job grid is really a grid and thus ought to have separation between columns as well as rows. The `.css` is exclusively dealing with the latter.

An improvement would be to use `<thead>`/`<th>` instead of `<div>` for the portlet headers, but in order to keep collapsability, the `id` needs to move from the `<table>` to the `<tbody>`, and I didn't see how to do that. Also, *ideally* the header needs to know how many columns the table has, as well, although `colspan="255"` seems to work acceptably in modern browsers. (I think this would allow all the inline styling to be dropped.)

Another non-obvious improvement, though a pretty minor cosmetic issue, is that there should be empty cells so that the last row of the jobs grid always has three cells.

> And remove all hard coded styles from this plugin

Sure, I don't claim what I did was exhaustive. 🙂 Just fixing the most obvious issue on our instance...

**Comment by mwoehlke-kitware** [comment]
Oh, and `height: auto;` is specifically to make the table(s) more compact. Not strictly necessary, but even as-is the number of jobs visible on screen at once is noticeably reduced by the proposed patch. Without `height: auto;`, it's reduced to something like half or less, which is getting to the point I  can see users being very unhappy with the change.

**Comment by TobiX** [comment]
Due to the awesome work by @mawinter69 in #328 this might get fixed rather sooner then later 😄 

Anyone want to test an incrementals build from https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/dashboard-view/2.508.v5a_8e0fa_0ffd2/?

**Comment by TobiX** [comment]
Should be fixed in [2.508.va_74654f026d1](https://github.com/jenkinsci/dashboard-view-plugin/releases/tag/2.508.va_74654f026d1), which was just released. Feel free to reopen (or create a new issue) if you still see problems.

**Comment by TobiX** [comment]
(Please be aware that some of the incremental releases leading up to this release might list as newer then the release so you if you installed one of those versions, you need to manually install the release version)

**Comment by svenevs** [comment]
Thank you for fixing this!

> (Please be aware that some of the incremental releases leading up to this release might list as newer then the release so you if you installed one of those versions, you need to manually install the release version)

Indeed, the UI plugin update got us to `2.495.v07e81500c3f2`.  Do you have an understanding of when that shows up there?  Unfortunately, I don't know how to manage the plugins on our instance manually and generally speaking don't have permission to test on production.

From my look at the PR there was a real over-haul / a lot of dedication to improving the style compatibility, thanks so much :slightly_smiling_face:

**Comment by TobiX** [comment]
Thank @mawinter69, not me 😉 

Anyways, [2.495.v07e81500c3f2](https://github.com/jenkinsci/dashboard-view-plugin/releases/tag/2.495.v07e81500c3f2) is the release from last August, you (or your admin) should be able to update to [2.508.va_74654f026d1](https://github.com/jenkinsci/dashboard-view-plugin/releases/tag/2.508.va_74654f026d1) via the normal Jenkins update center (if you are on a recent Jenkins LTS release - which you should be, otherwise your Jenkins instance is susceptible to security issues described in [Jenkins Security Advisory 2024-01-24](https://www.jenkins.io/security/advisory/2024-01-24/))

**Comment by svenevs** [comment]
True, excellent work @mawinter69! :partying_face: 

Hmmm, I guess this is my source of confusion.  We had actually fully updated our Jenkins last Friday 2024-02-02.  When I saw the traffic here I checked for plugin updates and that is when it gave me 2.495.v07e81500c3f2 via the UI update center.

I might be able to fiddle with it this weekend and try and set it up manually :slightly_smiling_face:  Thanks for the responses!
